lib/internal/fs/glob.js

```diff
@@ -376,6 +396,238 @@ class Glob {
       }
     }
   }
+
+  async *#addSubpatterns(path, pattern) {
```

Comment:
we should try to avoid this code duplication, perhaps extracting logic to common methods will allow that

Reply:
extracted in #48594